sdk/core/azure-core/src/main/java/com/azure/core/implementation/RestProxy.java

```diff
@@ -154,6 +159,36 @@ public Object invoke(Object proxy, final Method method, Object[] args) {
         }
     }
 
+    private Flux<ByteBuffer> validateLength(final HttpRequest request) {
+        final Flux<ByteBuffer> bbFlux = request.body();
+        if (bbFlux == null) {
+            return Flux.empty();
+        }
+
+        return Flux.defer(() -> {
+            Long expectedLength = Long.valueOf(request.headers().value("Content-Length"));
+            final long[] currentTotalLength = new long[1];
```

Comment:
I guess what we're doing here is correct. I asked the question in SOF and got a response from David (the main contributor of rx) - https://stackoverflow.com/questions/57776012/mutating-array-elements-in-reactive-publisher-onnext-doonnext-methods/
The last thing to rule out is whether to use volatile keyword to ensure immediate update visibility across thread, waiting for his response.

Reply:
we're good, David confirmed that framework takes care of memory visibility, no need for volatile. https://stackoverflow.com/questions/57776012/mutating-array-elements-in-reactive-publisher-onnext-doonnext-methods/57786588?noredirect=1#comment102032427_57786588

Reply:
Thanks Anu for taking care of this! Learned a lot from the thread.